src/error/RecordDuplicateError.ts

```diff
@@ -0,0 +1,7 @@
+import { AriesFrameworkError } from './AriesFrameworkError'
+
+export class RecordDuplicateError extends AriesFrameworkError {
+  public constructor(message: string, { recordType, cause }: { recordType: string; cause?: Error }) {
```

Comment:
I'm not sure about mixing domain info with infra info into one param `{ recordType, cause }`. On the other hand, I understand that if we have AriesFrameworkError(message, cause, details/metadata) or vice versa `AriesFrameworkError(message, details/metadata, cause)` we could end up with empty param for one case or another 🤔  🤷‍♂️

Reply:
Not sure how to act on this. Do you want me to change it?

My thinking was you have a message and all other stuff goes into the object. This way you don't have that problem with order of parameters

Reply:
Okay, I see. No, we don't have to change it. We can go this way and we'll see how it suits us.